### PR TITLE
Geome permit information

### PR DESCRIPTION
**Pull request by dannymandel**
Initial work for https://github.com/isamplesorg/isamples_inabox/issues/193

### CODE REVIEW

File: isamples_metadata/GEOMETransformer.py [hunk 1]
```diff
@@ -2,15 +2,20 @@
...
 import isamples_metadata
 from isamples_metadata.Transformer import (
     Transformer,
 )
 
+PERMIT_STRINGS_TO_IGNORE = ['nan', 'na', 'no data', 'unknown', 'none_required']
+
 TISSUE_ENTITY = "Tissue"
 JSON_TIME_FORMAT = "%Y-%m-%dT%H:%M:%S%z"
+PERMIT_STRUCTURED_TEXT_COMPLIES_WITH_PATTERN = re.compile(r"complies_with:([^\s]+)(\s+authorized_by:(.*))?")
```
Comment (datadavev) [inline isamples_metadata/GEOMETransformer.py]: Thinking about this a bit more- human entries are typically messy, especially during field work. Should probably also be prepared to accept tags like "compliesWith:", "complieswith:", maybe "complies with:", and perhaps even "complies:". Similarly for authorized_by.

File: tests/test_geome_permit_parsing.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        {"authorizedBy": [
+            "The SPICE project (Science for the protection of indonesian Coastal marine Ecosystems) was conducted and permitted under the governmental agreement between the German Federal Ministry of Education and Research (BMBF) and the Indonesian Ministry for Research and Technology (RISTEK), Indonesian Institute of Sciences (LIPI), Indonesian Ministry of Maritime Affairs and Fisheries (DKP), and Agency for the Assessment and Application of Technology (BPPT). This work was carried out in co-operation with Hassanuddin University (UNHAS, Makassar, Indonesia), Agricultural University Bogor (IPB, Bogor, Indonesia), and Jenderal Soedirman University (Purwokerto, Indonesia)."],
+            "compliesWith": []}),
+    ("DAFF/DEA", {"authorizedBy": ["DAFF", "DEA"], "compliesWith": []}),
+    ("Ministry of Environment, Natural Resources, and Transport",
```
Comment (datadavev) [inline tests/test_geome_permit_parsing.py:51]: This one should result in a single string "`Ministry of Environment, Natural Resources, and Transport`"

**Review by datadavev — APPROVED** [review]
Good first pass. We will likely need to be a bit flexible on the acceptable text, but that might be something better done after exercising the current impl and getting feedback.

File: isamples_metadata/GEOMETransformer.py (hunk 1 above)
Comment (dannymandel) [reply isamples_metadata/GEOMETransformer.py]: All of these should be working now -- I've tweaked the regexes and added tests.

File: tests/test_geome_permit_parsing.py (hunk 2 above)
Comment (dannymandel) [reply tests/test_geome_permit_parsing.py:51]: I'm not sure how smart we can be here without regressing some of the other ones.  I'll remove the comma rule and see how many tests break and whether there is a way to tighten up that rule.

**Pull request merged**